src/journal/bsod.c

```diff
@@ -194,13 +215,27 @@ static int display_emergency_message_fullscreen(const char *message) {
                 goto cleanup;
         }
 
+        if (message_id) {
+                r = loop_write(fd, "\nScan the QR code below, and search for the entry with the ID: ", SIZE_MAX);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to write emergency message to terminal: %m");
+
+                r = loop_write(fd, message_id, SIZE_MAX);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to write emergency message to terminal: %m");
+        }
+
         r = fdopen_independent(fd, "r+", &stream);
         if (r < 0) {
                 ret = log_error_errno(errno, "Failed to open output file: %m");
                 goto cleanup;
         }
 
-        r = print_qrcode_full(stream, "Scan the QR code", message, qr_code_start_row, qr_code_start_column, w.ws_col, w.ws_row);
+        r = print_qrcode_full(stream,
+                              "Scan the QR code", "https://cgit.freedesktop.org/systemd/systemd/plain/catalog/systemd.catalog",
```

Comment:
that repository is old. we are on github these days. And it's not really useful for users. i#d really not bother with an URL for now, we first need to set up something useful server side I guess.

Reply:
I think it would make sense to include the message id in the message. i.e. just suffix the message string with `(Message ID: …)` or so?